pkg/config/piped.go

```diff
@@ -969,14 +974,23 @@ func (n *NotificationReceiverSlack) Validate() error {
 	if len(mentionedAccounts) > 0 {
 		n.MentionedAccounts = mentionedAccounts
 	}
-	if n.HookURL != "" && (n.OAuthToken != "" || n.ChannelID != "") {
-		return errors.New("only one of hookURL or oauthToken and channelID should be used")
+	if n.HookURL != "" && (n.OAuthToken != "" || n.OAuthTokenFile != "" || n.OAuthTokenData != "" || n.ChannelID != "") {
+		return errors.New("only one of hookURL or OAuthToken, OAuthTokenData, OAuthTokenFile and channelID should be used")
 	}
 	if n.HookURL != "" {
 		return nil
 	}
-	if n.OAuthToken == "" || n.ChannelID == "" {
-		return errors.New("missing oauthToken or channelID configuration")
+	if (n.HookURL == "" && n.ChannelID == "") || (n.OAuthToken == "" && n.OAuthTokenFile == "" && n.OAuthTokenData == "") {
+		return errors.New("either HookURL or OAuthToken and OAuthTokenData, OAuthTokenFile and channelID must be set")
+	}
+	if n.HookURL == "" && n.OAuthToken == "" && n.OAuthTokenFile == "" && n.OAuthTokenData == "" {
+		return errors.New("either hookURL or OAuthToken and OAuthTokenData and OAuthTokenFile must be set")
+	}
+	if n.HookURL == "" && ((n.OAuthTokenData != "" || n.OAuthToken != "") && n.OAuthTokenFile != "") {
+		return errors.New("only one of OAuthToken or OAuthTokenData or OAuthTokenFile should be used")
+	}
+	if n.OAuthToken != "" && n.OAuthTokenData != "" {
+		return errors.New("only one of OAuthToken or OAuthTokenData should be used")
```

Comment:
```suggestion
	if n.OAuthToken != "" && n.OAuthTokenFile != "" && n.OAuthTokenData != "" {
		return errors.New("only one of OAuthToken, OAuthTokenData and OAuthTokenFile should be set")
```
Is this enough? 🤔

Reply:
well I think so too, a bit lazy think from my mind to check the actual need in this case, but this suggestion is correct
thx